pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -240,12 +241,30 @@ private List<String> getInstanceIds(String query, List<String> tableNames, Strin
     List<String> instanceIds;
     if (!tableNames.isEmpty()) {
       List<TableConfig> tableConfigList = getListTableConfigs(tableNames, database);
-      if (tableConfigList == null || tableConfigList.isEmpty()) {
-        throw QueryErrorCode.TABLE_DOES_NOT_EXIST.asException("Unable to find table in cluster, table does not exist");
+      List<LogicalTableConfig> logicalTableConfigList = null;
+      // First check for table configs, if not found, check for logical table configs.
+      if (tableConfigList.size() != tableNames.size()) {
+        logicalTableConfigList = getListLogicalTableConfigs(tableNames, database);
+        // If config is not found for all tables, then find the tables that are not found.
+        if ((tableConfigList.size() + logicalTableConfigList.size()) != tableNames.size()) {
```

Comment:
I presume `tableConfigList` + `logicalTableConfigList` can't have larger cardinality than `tableNames` because we enforce somewhere that logical tables can't have the same name as physical tables?